### PR TITLE
Add Solution homework - 12

### PR DESCRIPTION
**Pull request by BilkoYevgen**
Check my solution please

### CODE REVIEW

File: HW-12/main.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    elif len(split_plus[1]) != 12:
+        print('Length is not correct')
+        return False
+    else:
+        return True
+
+
+phonebook = {}
+
+with open('phonebook.json', 'r') as file:
```
Comment (klymenok) [inline HW-12/main.py]: Add here some handling if file does not exist

File: HW-12/main.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    # ------------------------------------------------------------------------------------------
+
+    if command == 'add':
+        name = input('Type name: ')
+        number = input('Type number: ')
+        if name not in phonebook:
+            if number_validation(number):
+                phonebook[name] = number
+                with open('phonebook.json', 'w') as file:
```
Comment (klymenok) [inline HW-12/main.py]: Change the modifier to `w+` to avoid an error if the file doesn't exist

File: HW-12/main.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        break
+
+
+# 2. Написати декоратор, який буде записувати в файл назву функції, яку він декорує, і писати час її виклику.
+
+def decorator(func):
+    def wrapper(*args, **kwargs):
+        print(f'Function name: {func.__name__}')
+        print(f'Function was called at: {time.strftime("%d/%m/%Y, %H:%M:%S")}')
+        return func
```
Comment (klymenok) [inline HW-12/main.py]: You should call the `func` and pass *args and **kwargs there. Otherwise, your decorator won't work

File: HW-12/main.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+calc()
+
+
+# 3. В попередньо написаний кастомний Exception додати запис помилки і час її виникнення у файл.
+
+class MyException(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+        with open('exception.txt', 'a') as file:
```
Comment (klymenok) [inline HW-12/main.py]: Change the modifier to `a+` to avoid an error if the file does not exist

**Review by klymenok — CHANGES_REQUESTED** [review]

File: HW-12/main.py (hunk 1 above)
Comment (BilkoYevgen) [reply HW-12/main.py]: Try/except added

File: HW-12/main.py (hunk 2 above)
Comment (BilkoYevgen) [reply HW-12/main.py]: Changed. I will study the topic again

File: HW-12/main.py (hunk 3 above)
Comment (BilkoYevgen) [reply HW-12/main.py]: Changed. I will study the topic again

File: HW-12/main.py (hunk 4 above)
Comment (BilkoYevgen) [reply HW-12/main.py]: Changed. I will study the topic again

File: HW-12/main.py (hunk 2 above)
Comment (klymenok) [reply HW-12/main.py]: Well, actually, it was my mistake. The `w` modifier works both for existing and non-existing files.

File: HW-12/main.py (hunk 4 above)
Comment (klymenok) [reply HW-12/main.py]: The same mistake for `a`: this modifier allows to work with non-existing files. The difference between `a` and `a+` is that  `a+` allows to read from file but not only write to it

**Review by klymenok — APPROVED** [review]
Looks good!

**Pull request merged**